modules/3d/include/opencv2/3d/depth.hpp

```diff
@@ -0,0 +1,502 @@
+     * @param CACHE_SRC The cache data for the srcFrame will be prepared.
+     * @param CACHE_DST The cache data for the dstFrame will be prepared.
+     * @param CACHE_ALL The cache data for both srcFrame and dstFrame roles will be computed.
+     * @param CACHE_DEPTH The frame will be generated from depth image
+     * @param CACHE_PTS The frame will be built from point cloud
+     */
+    enum
+    {
+      CACHE_SRC = 1, CACHE_DST = 2, CACHE_ALL = CACHE_SRC + CACHE_DST
+    };
+
+    /** Indicates what pyramid is to access using get/setPyramid... methods:
+    * @param PYR_IMAGE The pyramid of RGB images
+    * @param PYR_DEPTH The pyramid of depth images
+    * @param PYR_MASK  The pyramid of masks
+    * @param PYR_CLOUD The pyramid of point clouds, produced from the pyramid of depths
+    * @param PYR_DIX   The pyramid of dI/dx derivative images
+    * @param PYR_DIY   The pyramid of dI/dy derivative images
+    * @param PYR_TEXMASK The pyramid of textured masks
+    * @param PYR_NORM  The pyramid of normals
+    * @param PYR_NORMMASK The pyramid of normals masks
+    */
+    enum
+    {
+        PYR_IMAGE = 0, PYR_DEPTH = 1, PYR_MASK = 2, PYR_CLOUD = 3, PYR_DIX = 4, PYR_DIY = 5, PYR_TEXMASK = 6, PYR_NORM = 7, PYR_NORMMASK = 8,
+        N_PYRAMIDS
+    };
+
+    OdometryFrame() : ID(-1) { }
+    virtual ~OdometryFrame() { }
+
+    CV_WRAP static Ptr<OdometryFrame> create(InputArray image = noArray(), InputArray depth = noArray(),
+                                             InputArray  mask = noArray(), InputArray normals = noArray(), int ID = -1);
+
+    CV_WRAP virtual void setImage(InputArray  _image) = 0;
+    CV_WRAP virtual void getImage(OutputArray _image) = 0;
+    CV_WRAP virtual void setDepth(InputArray  _depth) = 0;
+    CV_WRAP virtual void getDepth(OutputArray _depth) = 0;
+    CV_WRAP virtual void setMask(InputArray  _mask) = 0;
+    CV_WRAP virtual void getMask(OutputArray _mask) = 0;
+    CV_WRAP virtual void setNormals(InputArray  _normals) = 0;
+    CV_WRAP virtual void getNormals(OutputArray _normals) = 0;
+
+    CV_WRAP virtual void setPyramidLevels(size_t _nLevels) = 0;
+    CV_WRAP virtual size_t getPyramidLevels(int pyrType) = 0;
+
+    CV_WRAP virtual void setPyramidAt(InputArray  _pyrImage, int pyrType, size_t level) = 0;
+    CV_WRAP virtual void getPyramidAt(OutputArray _pyrImage, int pyrType, size_t level) = 0;
+
+    CV_PROP int ID;
+};
+
+
+/** Base class for computation of odometry.
+ */
+class CV_EXPORTS_W Odometry
+{
+public:
+
+    /** A class of transformation*/
+    enum
+    {
+      ROTATION = 1, TRANSLATION = 2, RIGID_BODY_MOTION = 4
+    };
+
+    virtual ~Odometry() { }
+
+    /** Create a new Odometry object based on its name. Currently supported names are:
+    * "RgbdOdometry", "ICPOdometry", "RgbdICPOdometry", "FastICPOdometry".
+    * @param odometryType algorithm's name
+    */
+    CV_WRAP static Ptr<Odometry> createFromName(const String& odometryType);
```

Comment:
cv::String is deprecared. Use std::string instead.

Reply:
Fixed